lib/index.js

```diff
@@ -23,6 +23,7 @@ const rules = {
     'no-unexpected-wire-adapter-usages': require('./rules/no-unexpected-wire-adapter-usages'),
     'no-rest-parameter': require('./rules/no-rest-parameter'),
     'no-unknown-wire-adapters': require('./rules/no-unknown-wire-adapters'),
+    'no-uppercase-with-underscore-property-name': require('./rules/no-uppercase-with-underscore-property-name'),
```

Comment:
Instead of creating a new rule, I would suggest enhancing the existing `valid-api` because they overlap.
In order to avoid breaking changes, we should make the existing `valid-api` rule configurable:


Here is an example of an `.eslintrc` with the `valid-api` rule configured
```json
{
  "rules": {
    "valid-api": ["error", {
      "dissallowUnderscoreUppercaseMix": true
    }]
  }
}
```

The `dissallowUnderscoreUppercaseMix` should be set to `false` if the option is not set.